atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/atomic/ResilientCommitTimestampAtomicTable.java

```diff
@@ -86,7 +87,8 @@ public TransactionStatus load(@Nonnull CellInfo cellInfo) {
                         }
                         if (followUpAction == FollowUpAction.PUT) {
                             store.put(ImmutableMap.of(
-                                    cellInfo.cell(), encodingStrategy.transformStagingToCommitted(cellInfo.value())));
+                                    cellInfo.cell(),
+                                    TwoPhaseEncodingStrategy.INSTANCE.transformStagingToCommitted(cellInfo.value())));
```

Comment:
This pattern where we use one strategy then another seems a bit unusual to me. I think I'd prefer a design where the TwoPhaseEncodingStrategy wasn't a singleton and instead took in enough input to correctly differentiate behaviour in the V3 and V4 cases. Then we can just use that throughout.